### DISCUSSION TITLE
Instructions to build kitty with debug symbols?

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#4926] [Q&A] [answered]
I am running a recent nightly version of kitty and it was crashing very consistently in this very specific scenario I was running yesterday.

Before I dig deeper, I'd like to know how to build kitty with debug symbols and run it under gdb so that, hopefully, I can get a stack trace before filing a bug report.  (Of course, simply filing a bug without a stack trace isn't going to get us anywhere.)

The odd thing is that when kitty crashed, GNOME Shell (on my Ubuntu 20.04 LTS) box also crashed.  `/var/crash` then contained a GNOME Shell memdump.  I'm am not sure if it was GNOME Shell _that_ irritated kitty or vice versa.  Other running apps kept running—just kitty disappeared and GNOME Shell restarted.

**Comment by page-down** [comment]
I see the following in Makefile, maybe it's what you need?

```Makefile
debug:
    python3 setup.py build $(VVAL) --debug

debug-event-loop:
    python3 setup.py build $(VVAL) --debug --extra-logging=event-loop
```

**Comment by kovidgoyal** [comment, marked as answer]
make debug

or better

make asan

**Comment by salmankhilji** [comment]
Well...no more repro today, so can't file a bug :(
